pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -893,56 +891,64 @@ func (r *Builder) PopulatorVolumes(vmRef ref.Ref, annotations map[string]string,
 		}
 		images = append(images, image)
 	}
-
 	for _, image := range images {
-
 		if image.Status != string(ImageStatusActive) {
 			r.Log.Info("the image is not ready yet", "image", image.Name)
 			continue
 		}
-
-		originalVolumeDiskId := image.Name
-		if imageProperty, ok := image.Properties[forkliftPropertyOriginalVolumeID]; ok {
-			originalVolumeDiskId = imageProperty.(string)
+		if pvc, pvcErr := r.getCorrespondingPvc(image, workload, vmRef, annotations, secretName); pvcErr == nil {
+			pvcs = append(pvcs, pvc)
+		} else {
+			err = pvcErr
+			return
 		}
+	}
+	return
+}
 
-		_, err = r.getVolumePopulator(image.Name)
-		if err != nil {
-			if !k8serr.IsNotFound(err) {
-				err = liberr.Wrap(err)
-				return
-			}
-		}
+func (r *Builder) getCorrespondingPvc(image model.Image, workload *model.Workload, vmRef ref.Ref, annotations map[string]string, secretName string) (*core.PersistentVolumeClaim, error) {
+	if populatorName, err := r.ensureVolumePopulator(workload, &image, secretName); err == nil {
+		return r.ensureVolumePopulatorPVC(workload, &image, annotations, populatorName)
+	} else {
```

Comment:
if used named return value we can dropped the else here

Reply:
but with named return values, I'll need to define populatorName separately... I can drop the 'else' now as well..

Reply:
but err will be out of scope (unless defined at the top), no?

Reply:
ah right right

Reply:
changed, also resolved a conflict with #591 - please double check me